server.go

```diff
@@ -1519,8 +1523,17 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 		return nil, err
 	}
 
-	// Assemble a peer notifier which will provide clients with subscriptions
-	// to peer online and offline events.
+	if s.cfg.ProtocolOptions.OptionPeerStorage {
+		s.peerStorageProvider, err = peer.NewPeerStorageProducer(
```

Comment:
I think what might be cleaner is to just initiate the DB & the producers no matter what. Then we pass in the `s.cfg.ProtocolOptions.OptionPeerStorage` to the peer as a `SupportsStorage` boolean. Then in the peer message handlers, we then check this boolean instead of doing nil checks. ie, let's avoid nil checks where possible. 

interested to hear what @ProofOfKeags thinks too

Reply:
No need for the boolean, we can just check for the feature bit.